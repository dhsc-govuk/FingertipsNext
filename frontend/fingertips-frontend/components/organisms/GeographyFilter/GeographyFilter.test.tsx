import { render, screen } from '@testing-library/react';
import { GeographyFilter } from '.';

const availableAreaTypes = [
  {
    id: '001',
    name: 'area type 001',
  },
  {
    id: '002',
    name: 'area type 002',
  },
];

describe('Geography Filter', () => {
  it('snapshot test', () => {
    const container = render(<GeographyFilter availableAreaTypes={[]} />);

    expect(container.asFragment()).toMatchSnapshot();
  });

  it('should render the Filters heading', () => {
    render(<GeographyFilter />);

    expect(screen.getByRole('heading')).toHaveTextContent('Filters');
  });

  it('should render the selected areas and not the area type drop down when there are areas selected', () => {
    render(
      <GeographyFilter
        selectedAreas={[{ id: '001', name: 'selected area 001' }]}
      />
    );

    expect(screen.getByText(/Areas Selected/i)).toBeInTheDocument();
    expect(
      screen.queryByRole('combobox', { name: /Select an area type/i })
    ).not.toBeInTheDocument();
  });

  it('should render the select area type drop down if there are no selected areas', () => {
    render(<GeographyFilter availableAreaTypes={availableAreaTypes} />);

    expect(screen.queryByText(/Areas Selected/i)).not.toBeInTheDocument();
    expect(
      screen.getByRole('combobox', { name: /Select an area type/i })
    ).toBeInTheDocument();
  });

  it('should render all the available areaTypes', () => {
    render(<GeographyFilter availableAreaTypes={availableAreaTypes} />);

    expect(screen.queryByText(/Areas Selected/i)).not.toBeInTheDocument();
    expect(
      screen.getByRole('combobox', { name: /Select an area type/i })
    ).toHaveLength(2);
  });
});
