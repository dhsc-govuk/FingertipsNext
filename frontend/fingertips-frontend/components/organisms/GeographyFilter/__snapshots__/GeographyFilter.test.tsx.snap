// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Geography Filter snapshot test 1`] = `
<DocumentFragment>
  .c5 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  display: block;
  clear: none;
  padding-bottom: 2px;
  margin-bottom: 0;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  margin-bottom: 0;
}

.c3:after {
  content: '';
  display: table;
  clear: both;
}

.c1 {
  color: #0b0c0c;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  font-size: 18px;
  line-height: 1.1111111111111112;
  display: block;
  margin-top: 0;
  margin-bottom: 15px;
}

.c2 {
  margin: 0;
  border: 0;
  border-bottom: 1px solid #bfc1c3;
}

.c6 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  box-sizing: border-box;
  width: 100%;
  height: 33px;
  padding: 5px 4px 4px;
  border: 2px solid #0b0c0c;
}

.c6:focus {
  outline: 3px solid #ffdd00;
  outline-offset: 0;
}

.c0 {
  background-color: #E1E2E3;
  min-height: 100%;
  padding: 0.5em;
}

.c4 select {
  width: 100%;
}

@media print {
  .c5 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c5 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c5 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c5 {
    margin-bottom: 0;
  }
}

@media only screen and (min-width:641px) {
  .c3 {
    margin-bottom: 0;
  }
}

@media print {
  .c1 {
    color: #000;
  }
}

@media print {
  .c1 {
    font-size: 18px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    font-size: 24px;
    line-height: 1.25;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    margin-bottom: 20px;
  }
}

@media print {
  .c6 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c6 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c6 {
    width: 50%;
    height: 38px;
  }
}

<div
    class="c0"
    data-testid="geography-filter-container"
  >
    <h3
      class="c1"
    >
      Filters
    </h3>
    <hr
      class="c2"
    />
    <label
      class="c3 c4"
    >
      <span
        class="c5"
      >
        Select an area type
      </span>
      <select
        class="c6"
      />
    </label>
  </div>
</DocumentFragment>
`;
